rust/src/tests/mknod.rs

```diff
@@ -52,3 +61,106 @@ fn changed_time_fields_success(ctx: &mut TestContext) {
             mknod_wrapper(&path, Mode::from_bits_truncate(0o644)).unwrap();
         });
 }
+
+#[cfg(any(target_os = "linux", target_os = "freebsd", target_os = "solaris"))]
+crate::test_case! {
+    /// mknod create device files
+    // mknod/11.t
+    device_files, root => [Block, Char]
+}
+#[cfg(any(target_os = "linux", target_os = "freebsd", target_os = "solaris"))]
+fn device_files(ctx: &mut TestContext, ft: FileType) {
+    use crate::utils::dev::{major, makedev, minor};
+    use nix::sys::stat::mode_t;
+
+    let (argument, check): (SFlag, fn(&StdFileType) -> bool) = match ft {
+        FileType::Block => (SFlag::S_IFBLK, StdFileType::is_block_device),
+        FileType::Char => (SFlag::S_IFCHR, StdFileType::is_char_device),
+        _ => unreachable!(),
+    };
+
+    let file = ctx.gen_path();
+
+    let mode = 0o755;
+    let major_num = 1;
+    let minor_num = 2;
+
+    assert!(mknod(
+        &file,
+        argument,
+        Mode::from_bits_truncate(mode),
+        makedev(major_num, minor_num)
+    )
+    .is_ok());
+
+    let stat = symlink_metadata(&file).unwrap();
+    assert_eq!(stat.permissions().mode() as mode_t & ALLPERMS, mode);
+    assert_eq!(major(stat.rdev()) as u64, major_num as u64);
+    assert_eq!(minor(stat.rdev()) as u64, minor_num as u64);
+    assert!(check(&stat.file_type()));
+
+    assert_eq!(
```

Comment:
The EEXIST stuff is totally redundant with what's in 08.t.  I suggest deleting it here.  If you do keep it, you should move it to a separate test function.